packages/ag-charts-community/src/scene/clip.ts

```diff
@@ -0,0 +1,52 @@
+import { jsonDiff } from '../util/json';
+import { BBox } from './bbox';
+import { Node, RedrawType, type RenderContext } from './node';
+
+type Constructor<T> = new (...args: any[]) => T;
+type Circle = { x: number; y: number; radius: number };
+
+export type ClippableOutsideType<T> = T & {
+    clipShape?: BBox | Circle;
+    setClipMask: (shape?: BBox | Circle) => void;
+};
+
+export function ClippableOutside<N extends Node>(Parent: Constructor<N>): Constructor<ClippableOutsideType<N>> {
```

Comment:
I looked into doing this as a general `Clippable` mixin with a mode of `inside | outside`. For inside, I overrode the render method with:

```ts
ctx.globalCompositeOperation = 'source-in'
super.render(renderCtx);
ctx.globalCompositeOperation = prevOperation;
```

But this impacts all the other nodes within the same canvas layer, erasing them.

If we end up with a use-case for this approach we can look into this further.

Reply:
Ah, so maybe you need to use `ctx.save()` and `ctx.restore()` rather than manually managing the ctx state rollback here? I think you need to use this approach anyway to avoid state leakage?

Agree we can leave this use-case for now.